sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/implementation/caches/AsyncCacheNonBlockingTest.java

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.cosmos.implementation.caches;
+
+import com.azure.cosmos.implementation.guava25.base.Function;
+import org.testng.annotations.Test;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.stream.Collectors;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class AsyncCacheNonBlockingTest {
+    private static final int TIMEOUT = 2000;
+
+    @Test(groups = {"unit"}, timeOut = TIMEOUT)
+    public void getAsync() {
+        AtomicInteger numberOfCacheRefreshes = new AtomicInteger(0);
+        final Function<Integer, Mono<Integer>> refreshFunc = key -> {
```

Comment:
let `numberOfCacheRefreshes.incrementAndGet();` be part of the Mono, you will see the issue mentioned above about `I think we probably need to cache the result here, turning into a hot publisher, else for each subscribe, it will trigger a call to gateway?`